packages/dotenv-webpack-plugin/src/index.js

```diff
@@ -0,0 +1,148 @@
+const dotenv = require("dotenv");
+const dotenvExpand = require("dotenv-expand");
+const { DefinePlugin } = require("webpack");
+const { validate } = require("schema-utils");
+const schema = require("./options.json");
+
+/** @typedef {import("./types").Config} Config */
+/** @typedef {import("schema-utils/declarations/validate").Schema} Schema */
+/** @typedef {import("webpack").Compiler} Compiler */
+
+class DotenvWebpackPlugin {
+  /**
+   * Dotenv Webpack Plugin
+   * @param {Config} options - Configuration options
+   */
+  constructor(options = {}) {
+    validate(/** @type {Schema} */ (schema), options || {}, {
+      name: "DotenvWebpackPlugin",
+      baseDataPath: "options",
+    });
+
+    const currentDirectory = process.cwd();
+
+    const {
+      envFiles = [
+        `${currentDirectory}/.env.example`, // loaded in all cases
+        `${currentDirectory}/.env`, // loaded in all cases
+        `${currentDirectory}/.env.local`, // loaded in all cases, ignored by git
+        `${currentDirectory}/.env.[mode]`, // only loaded in specified mode
+        `${currentDirectory}/.env.[mode].local`, // only loaded in specified mode, ignored by git
```

Comment:
Let's make order like in next.js https://nextjs.org/docs/app/building-your-application/configuring/environment-variables.

Reply:
Also let's add test cases on order too

Reply:
Changed order to be same as NextJS and added a test case that asserts on the order